src/Shared/TypeLoader.cs

```diff
@@ -317,31 +327,190 @@ internal LoadedType GetLoadedTypeByTypeName(string typeName)
+                        if (!typeDef.Attributes.HasFlag(TypeAttributes.Public) || !typeDef.Attributes.HasFlag(TypeAttributes.Class))
+                        {
+                            continue;
+                        }
+                        else
+                        {
+                            string currentTypeName = metadataReader.GetString(typeDef.Name);
+                            if (currentTypeName.Length == 0 || TypeLoader.IsPartialTypeNameMatch(currentTypeName, typeName))
+                            {
+                                // We found the right type! Now get its information.
+                                foreach (CustomAttributeHandle customAttrHandle in typeDef.GetCustomAttributes())
+                                {
+                                    CustomAttribute customAttribute = metadataReader.GetCustomAttribute(customAttrHandle);
+                                    MemberReference constructorReference = metadataReader.GetMemberReference((MemberReferenceHandle)customAttribute.Constructor);
+                                    if (constructorReference.Parent.Kind == HandleKind.TypeReference)
+                                    {
+                                        TypeReference typeReference = metadataReader.GetTypeReference((TypeReferenceHandle)constructorReference.Parent);
+                                        string customAttributeName = metadataReader.GetString(typeReference.Name);
+                                        switch (customAttributeName)
+                                        {
+                                            case "RunInSTAAttribute":
+                                                typeInformation.HasSTAThreadAttribute = true;
+                                                break;
+                                            case "LoadInSeparateAppDomainAttribute":
+                                                typeInformation.HasLoadInSeparateAppDomainAttribute = true;
+                                                break;
+                                        }
+                                    }
+                                }
+
+                                IEnumerable<PropertyDefinition> propertyDefinitions = typeDef.GetProperties().Select(prop => metadataReader.GetPropertyDefinition(prop));
+                                List<TypeInformationPropertyInfo> typePropertyInfos = new();
+                                foreach (PropertyDefinition propertyDefinition in propertyDefinitions)
+                                {
+                                    TypeInformationPropertyInfo toAdd = new();
+                                    toAdd.Name = metadataReader.GetString(propertyDefinition.Name);
+                                    byte[] bytes = metadataReader.GetBlobReader(propertyDefinition.Signature).ReadBytes(metadataReader.GetBlobReader(propertyDefinition.Signature).Length);
```

Comment:
bytes isn't currently used. I need to get type information for each property, and that's unusually complicated in the general case because of generics and custom types. We may be able to get by without them because there are only so many things that can be passed as inputs/outputs from a task—thoughts @rainersigwald? Do I need a custom SignatureDecoder? (I was also considering using propertyDefinition.DecodeSignature, but I wasn't sure what TType or TGenericContext would be.)

Reply:
Specifically, acceptable types include value types, strings, ITaskItems, and arrays of any of those.

Reply:
Should we really be accepting structs? Maybe we should require value types to be { integer, float, char, bool, nullable of one of those, or enum }?